src/ld/Dockerfile

```diff
@@ -64,42 +101,60 @@ RUN export CFLAGS=${LIPO_CFLAGS} && \
 FROM scratch AS lipo-static
 COPY --from=lipo-base /work/cctools-port/cctools/misc/lipo /
 
-FROM cctools-base AS codesign-base
+FROM cctools-base AS codesign-allocate-base
 ARG CODESIGN_CFLAGS="-Wl,-s -Os"
 RUN export CFLAGS=${CODESIGN_CFLAGS} && \
     ./configure --host=$(xx-clang --print-target-triple) LDFLAGS=--static && \
     make -C libmacho && make -C libstuff && make -C misc codesign_allocate && \
     xx-verify --static misc/codesign_allocate
 
-FROM scratch AS codesign-static
-COPY --from=codesign-base /work/cctools-port/cctools/misc/codesign_allocate /
-
-FROM cctools-base AS otool-base
-ARG OTOOL_CFLAGS="-Wl,-s -Os"
-RUN export CFLAGS=${OTOOL_CFLAGS} && \
-    ./configure --host=$(xx-clang --print-target-triple) LDFLAGS=--static && \
-    make -C libstuff && make -C libobjc2 && make -C otool && \
-    xx-verify --static otool/otool
-
-FROM scratch AS otool-static
-COPY --from=otool-base /work/cctools-port/cctools/otool/otool /
+FROM scratch AS codesign-allocate-static
+COPY --from=codesign-allocate-base /work/cctools-port/cctools/misc/codesign_allocate /
 
 FROM cctools-base AS ld64-static-base
 # for LTO
 RUN apk add llvm-dev
 ARG LD64_CXXFLAGS="-Wl,-s -Os"
 RUN --mount=from=libtapi-static,source=/opt/libtapi,target=/opt/libtapi \
     export CXXFLAGS=${LD64_CXXFLAGS} && ./configure --with-libtapi=/opt/libtapi --host=$(xx-clang --print-target-triple) &&\
-    sed -i 's/-ltapi/-ltapi -ltapiCore -lLLVMObject -lLLVMSupport -lLLVMDemangle -lLLVMMC -lLLVMBinaryFormat --static/' ld64/src/ld/Makefile && \
+    sed -i 's/-ltapi/-ltapi -ltapiCore -ltapiObjCMetadata -lLLVMObject -lLLVMTextAPI -lLLVMSupport -lLLVMDemangle -lLLVMMC -lLLVMBinaryFormat --static/' ld64/src/ld/Makefile && \
     make -j $(nproc) -C ld64 && \
     xx-verify --static ld64/src/ld/ld
 
+FROM --platform=${BUILDPLATFORM} alpine:${ALPINE_VERSION} AS sigtool-base
```

Comment:
Because of this wrong stage order the latest prebuilt ld64 version is missing these files and broken

Reply:
:disappointed: